src/main/scala/com/databricks/spark/redshift/Parameters.scala

```diff
@@ -242,5 +256,26 @@ private[redshift] object Parameters extends Logging {
       ((s"$hadoopConfPrefix.awsAccessKeyId", accessKeyId),
         (s"$hadoopConfPrefix.awsSecretAccessKey", secretAccessKey))
     }
+
+    def validateS3Configuration(configuration: Configuration): Unit = {
+      if (!disableS3LifecycleCheck) {
+        val ((_, accessKeyId), (_, secretAccessKey)) = credentialsTuple(configuration)
+        val s3Client = new AmazonS3Client(new BasicAWSCredentials(accessKeyId, secretAccessKey))
+        val s3URI = new AmazonS3URI(Utils.fixS3Url(tempDir))
+        val bucket = s3URI.getBucket
+        val bucketLifecycleConfiguration = s3Client.getBucketLifecycleConfiguration(bucket)
+        val key = s3URI.getKey
+        val someRuleMatchesTempDir = bucketLifecycleConfiguration.getRules.asScala.exists { rule =>
+          rule.getStatus == BucketLifecycleConfiguration.ENABLED && key.startsWith(rule.getPrefix)
```

Comment:
Minor subtlety: this only checks that there's an active matching rule, not that the rule actually enforces deletion. Still, this is better than nothing.